vmdb/app/models/ems_kubernetes.rb

```diff
@@ -11,19 +11,40 @@ def self.description
     @description ||= "Kubernetes".freeze
   end
 
-  def self.raw_connect(hostname, port, api_version)
+  def self.raw_connect(uri, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    Kubeclient::Client.new(uri, api_version)
+  end
 
+  def api_endpoint
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
+    uri = URI::HTTP.build(:path => '/api', :port => port.to_i)
     uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    uri.to_s
   end
 
   def connect(_options = {})
-    self.class.raw_connect(hostname, port, api_version)
+    raw_connect(api_endpoint, api_version)
```

Comment:
How does this work if raw_connect is a class method? (note, it should stay a class method for consistency and for diagnosis).